pkg/vm/engine/tae/common/dlnode.go

```diff
@@ -208,9 +208,13 @@ func (it *LinkIt) Valid() bool {
 }
 
 func (it *LinkIt) Next() {
-	it.linkLocker.RLock()
-	it.curr = it.nextFunc(it.curr)
-	it.linkLocker.RUnlock()
+	if it.linkLocker != nil {
```

Comment:
There are so many questions regarding this little library.

Who is this lock guarding?  Obviously INSERT or DELETE to the list does not acquire this lock.   So, what is the point?